lib/absinthe/phase/document/execution/resolution.ex

```diff
@@ -75,11 +78,35 @@ defmodule Absinthe.Phase.Document.Execution.Resolution do
   end
 
   defp run_callbacks(plugins, callback, acc, true) do
-    Enum.reduce(plugins, acc, &apply(&1, callback, [&2]))
+    Enum.reduce(plugins, acc, fn plugin, acc ->
+      start_time = System.monotonic_time()
+
+      emit_start_event(start_time, acc, plugin)
```

Comment:
Very close! So, `callback` here is either the atom `:before_resolution` or `:after_resolution`, and I was thinking we could use that to dispatch more finely tuned events. Specifically this would get us:

```
[:absinthe, :plugin, :before_resolution, :start]
[:absinthe, :plugin, :before_resolution, :stop]
[:absinthe, :plugin, :after_resolution, :start]
[:absinthe, :plugin, :after_resolution, :stop]
```